fe/fe-core/src/main/java/org/apache/doris/backup/BackupHandler.java

```diff
@@ -621,8 +681,16 @@ public static BackupHandler read(DataInput in) throws IOException {
     public void write(DataOutput out) throws IOException {
         repoMgr.write(out);
 
-        out.writeInt(dbIdToBackupOrRestoreJob.size());
-        for (AbstractJob job : dbIdToBackupOrRestoreJob.values()) {
+        jobLock.lock();
```

Comment:
Actually, we don't need to lock here, cause the `write()` method can only be called in Checkpoint thread, so there is no concurrency issue. 
And by the way, I think we can just use:
`for (AbstractJob job : dbIdToBackupOrRestoreJob.values())`